Zend/zend_compile.c

```diff
@@ -3771,20 +3775,71 @@ void zend_compile_static_call(znode *result, zend_ast *ast, uint32_t type) /* {{
 }
 /* }}} */
 
-void zend_compile_class_decl(zend_ast *ast);
+void zend_compile_anon_use(znode *object, zend_ast *use_ast) { /* {{{ */
+	zend_ast *use_ast_list = use_ast->child[0];
+	zend_ast_list *use_list = zend_ast_get_list(use_ast_list);
+	uint32_t i = 0;
+
+	for (i = 0; i < use_list->children; i++) {
+		zend_ast *use_var = use_list->child[i];
+		zend_op *opline;
+		zend_bool by_ref = use_var->attr;
+
+		switch (use_var->kind) {
+			case ZEND_AST_ZVAL: {
+				zend_string *var_name = 
+					zend_string_copy(
+						zend_ast_get_str(use_var));
+
+				if (zend_string_equals_literal(var_name, "this")) {
+					zend_error_noreturn(E_COMPILE_ERROR, "Cannot use $this as lexical variable");
+				}
+
+				if (zend_is_auto_global(var_name)) {
+					zend_error_noreturn(E_COMPILE_ERROR, "Cannot use auto-global as lexical variable");
+				}
```

Comment:
Solved for use($a, $a), but properties are a bit difficult to detect duplicates of, I think ...